### ISSUE TITLE
io->catfile( $io, $str ) fails on windows

### ISSUE THREAD

**Issue by ediblenergy** [ingydotnet/io-all-pm#74]
```
use strict;                              
use warnings FATAL => 'all';             

BEGIN {                                  
    $^O = 'MSWin32';                     
};                                       

use IO::All;                             
io->catfile( io->dir('/'), 'test.json' );

Operation "eq": no method found,
    left argument in overloaded package IO::All::Dir,
    right argument has no overloaded magic at /home/skaufman/perl5/perlbrew/perls/perl-5.20.0/lib/5.20.0/x86_64-linux/File/Spec/Win32.pm line 135.
```

Also see: http://www.cpantesters.org/cpan/report/608781a5-9264-1014-8cee-01022b825c07
For when that bug shows up from another package.

Googling around I found: http://www.nntp.perl.org/group/perl.beginners/2005/04/msg74371.html
Which somewhat suggests that overloading "" is not enough, you may have to overload 'eq' as well. That seems odd to me, but maybe.

**Comment by ediblenergy** [comment]
skaufman@yibberdibber:~ $ perl -MFile::Spec::Win32 -e 'warn File::Spec::Win32->VERSION'
    3.47 at -e line 1.

**Comment by ediblenergy** [comment]
adding use overload 'eq' => '_overload_stringify'; to IO::All fixes it, haarg suggested adding a fallback to _overload_stringify.

**Comment by xlat** [comment]
Would'nt it be better to add a cmp overload?

**Comment by ediblenergy** [comment]
@xlat as long as it fixes the 'Operation "eq": no method found' I'm happy. The answer to your question as far as I'm concerned is if overloading cmp overloads eq by proxy then sure, that would be better.

**Comment by xlat** [comment]
yes, it looks like to be the case if you take a look at the second test I have added to overload.t ;)

**Comment by ediblenergy** [comment]
Ah cool now I see it, looks good to me :) I'll ping in #io-all because github notifications are terrible.

**Comment by frioux** [comment]
I'll try to get a fix released for this soonish; thanks for the research all!

**Comment by wesQ3** [comment]
I have this issue as well on Windows using `->relative($path)`.

```
perl -MCarp::Always -E "use IO::All; my $p = io->dir('C:\Inetpub\log');
 my $f =  $p->file('06-30-2017.txt'); say $f->relative($p)"

Operation "eq": no method found,
        left argument in overloaded package IO::All::Dir,
        right argument has no overloaded magic at C:/Inetpub/lynx/strawberry/perl/site/lib/File/Spec/Win32.pm line 186.
        File::Spec::Win32::canonpath("File::Spec", IO::All::Dir=GLOB(0x3faf3c)) called at C:/Inetpub/lynx/strawberry/perl/site/lib/File/Spec/Unix.pm line 414
        File::Spec::Unix::abs2rel("File::Spec", "..\\..\\log\\06-30-2017.txt", IO::All::Dir=GLOB(0x3faf3c)) called at C:/Inetpub/lynx/strawberry/perl/site/lib/IO/All/Filesys.pm line 77
        IO::All::Filesys::relative(IO::All::File=GLOB(0x228a354), IO::All::Dir=GLOB(0x3faf3c)) called at -e line 1
```

**Comment by frioux** [comment]
Wes told me he'd write a test; I'll incorporate one of the fixes above, have him ensure that the test passes on win32 (and ofc I'll run on linux) and then I'll cut a rel.

**Comment by wesQ3** [comment]
This is resolved in the 0.87 release and can be closed.
